ecl/resource_ecl_network_load_balancer_v2.go

```diff
@@ -0,0 +1,1384 @@
+	}
+
+	log.Printf("[DEBUG] Retrieved Load Balancer %s: %+v", d.Id(), loadBalancer)
+
+	d.Set("admin_username", loadBalancer.AdminUsername)
+	d.Set("availability_zone", loadBalancer.AvailabilityZone)
+	d.Set("default_gateway", loadBalancer.DefaultGateway)
+	d.Set("description", loadBalancer.Description)
+	d.Set("load_balancer_plan_id", loadBalancer.LoadBalancerPlanID)
+	d.Set("name", loadBalancer.Name)
+	d.Set("status", loadBalancer.Status)
+	d.Set("tenant_id", loadBalancer.TenantID)
+	d.Set("user_username", loadBalancer.UserUsername)
+
+	// Get Interface List and Call Show API to write back to ResourceData
+	for i, e := range loadBalancer.Interfaces {
+		lbInterface, err := load_balancer_interfaces.Get(networkClient, e.ID).Extract()
+		if err != nil {
+			return err
+		}
+		loadBalancer.Interfaces[i] = *lbInterface
+	}
+	d.Set("interfaces", flattenLoadBalancerInterfaces(loadBalancer.Interfaces))
+
+	// Get Syslog Server List and Call Show API to write back to ResourceData
+	for i, e := range loadBalancer.SyslogServers {
+		syslogServer, err := load_balancer_syslog_servers.Get(networkClient, e.ID).Extract()
+		if err != nil {
+			return err
+		}
+		loadBalancer.SyslogServers[i] = *syslogServer
+	}
+	syslogStates := d.Get("syslog_servers").([]interface{})
+	d.Set("syslog_servers", flattenLoadBalancerSyslogServers(loadBalancer.SyslogServers, syslogStates))
+
+	return nil
+}
+
+func resourceNetworkLoadBalancerV2Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	networkClient, err := config.networkV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL network client: %s", err)
+	}
+
+	// Retrieve Load Balancer Interface information (Interface IDs are required to update unused interfaces and plan)
+	loadBalancer, err := load_balancers.Get(networkClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "load-balancer")
+	}
+
+	// detect network-related changes
+	o, _ := d.GetChange("default_gateway")
+	oldDefaultGateway := o.(string)
+	gatewayHasChange := d.HasChange("default_gateway")
+
+	planHasChange := d.HasChange("load_balancer_plan_id")
+	interfaceHasChange := d.HasChange("interfaces")
+
+	syslogIPHasChange := false
+	o, _ = d.GetChange("syslog_servers.#")
+	oldSyslogServerNumber := o.(int)
+	for i := 0; i < oldSyslogServerNumber; i++ {
+		if d.HasChange(fmt.Sprintf("syslog_servers.%d.ip_address", i)) {
+			syslogIPHasChange = true
+			break
+		}
+	}
+
+	gatewayInitialized := false
+	planUpdated := false
+	syslogInitialized := false
+
+	// update interfaces
+	if interfaceHasChange {
+		if planHasChange {
+			// When Load Balancer Plan has been changed and interfaces are increased,
+			// update Load Balancer Plan before connect new interfaces.
+			// Max number of Load Balancer Interface could be exceeded, so the new Load Balancer Plan is needed.
+			_, n := d.GetChange("interfaces")
+			if len(loadBalancer.Interfaces) < len(n.([]interface{})) {
```

Comment:
ただ単にGetで良いと思います。またif文の中で取得すべきです。
```suggestion
			if len(loadBalancer.Interfaces) < len(d.Get("interfaces").([]interface{})) {
```

Reply:
別件との兼ね合いで、結果的に別の修正方法となりました。

loadBalancer.Interfacesについては、代わりにStateを使用するように別件で修正を加えました。
そのため、 `_, n := d.GetChange("interfaces")` の両方の戻り値が必要となり、
`len(o.([]interface{})) < len(n.([]interface{}))` という比較に修正しています。